beater/handlers.go

```diff
@@ -133,6 +134,18 @@ func isAuthorized(req *http.Request, secretToken string) bool {
 	return subtle.ConstantTimeCompare([]byte(parts[1]), []byte(secretToken)) == 1
 }
 
+func optionsHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "OPTIONS" {
+			w.Header().Set("Access-Control-Allow-Origin", "*")
+			w.Header().Set("Access-Control-Allow-Methods", "POST")
+			sendStatus(w, r, 200, nil)
+		} else {
+			h.ServeHTTP(w, r)
+		}
+	})
+}
```

Comment:
The `OPTION` request is also a request, just with a different http method.
Therefore I suggest to move the differentiation to the `processRequest` or the `processRequestHandler` method.

Reply:
yes, but i only want it for frontend, so i need a different handler that i can chain only for frontend endpoints